virtualizarr/readers/tiff.py

```diff
@@ -1,82 +1,197 @@
-            reader_options = {}
+    @staticmethod
+    async def _open_tiff(*, path: str, store: ObjectStore) -> TIFF:
+        from async_tiff import TIFF
+
+        return await TIFF.open(path, store=store)
 
-        reader_options.pop("storage_options", {})
-        warnings.warn(
-            "storage_options have been dropped from reader_options as they are not supported by kerchunk.tiff.tiff_to_zarr",
-            UserWarning,
+    @staticmethod
+    def _construct_manifest_array(
+        *, ifd: ImageFileDirectory, path: str
+    ) -> ManifestArray:
+        if not ifd.tile_height or not ifd.tile_width:
+            raise NotImplementedError(
+                f"TIFF reader currently only supports tiled TIFFs, but {path} has no internal tiling."
+            )
+        chunks = (ifd.tile_height, ifd.tile_height)
+        shape = (ifd.image_height, ifd.image_width)
+        chunk_manifest = TIFFVirtualBackend._construct_chunk_manifest(
+            ifd, path=path, shape=shape, chunks=chunks
         )
+        codecs = [_get_codecs(ifd.compression)]
+        codec_configs = [
+            numcodec_config_to_configurable(codec.get_config()) for codec in codecs
+        ]
+        dimension_names = ("y", "x")  # Folllowing rioxarray's behavior
 
-        # handle inconsistency in kerchunk, see GH issue https://github.com/zarr-developers/VirtualiZarr/issues/160
-        refs = KerchunkStoreRefs({"refs": tiff_to_zarr(filepath, **reader_options)})
+        metadata = create_v3_array_metadata(
+            shape=shape,
+            data_type=_get_dtype(
+                sample_format=ifd.sample_format, bits_per_sample=ifd.bits_per_sample
+            ),
+            chunk_shape=chunks,
+            fill_value=None,  # TODO: Fix fill value
+            codecs=codec_configs,
+            dimension_names=dimension_names,
+        )
+        return ManifestArray(metadata=metadata, chunkmanifest=chunk_manifest)
 
-        # both group=None and group='' mean to read root group
+    @staticmethod
+    def _construct_manifest_group(
+        store: ObjectStore,
+        path: str,
+        *,
+        group: str | None = None,
+    ) -> ManifestGroup:
+        """
+        Construct a virtual Group from a tiff file.
+        """
+        # TODO: Make an async approach
+        tiff = sync(TIFFVirtualBackend._open_tiff(store=store, path=path))
+        attrs: dict[str, Any] = {}
+        manifest_dict = {}
         if group:
-            refs = extract_group(refs, group)
+            manifest_dict[group] = TIFFVirtualBackend._construct_manifest_array(
+                ifd=tiff.ifds[int(group)], path=path
+            )
+        else:
+            for ind, ifd in enumerate(tiff.ifds):
+                manifest_dict[str(ind)] = TIFFVirtualBackend._construct_manifest_array(
+                    ifd=ifd, path=path
+                )
+        return ManifestGroup(manifest_dict=manifest_dict, attributes=attrs)
+
+    @staticmethod
+    def open_virtual_zarr(
```

Comment:
Let's call this something else, perhaps this?
```suggestion
    def create_manifeststore(
```